rero/rero-ils
# [spike] holdings: introduce holding level

* Adds holdings level between documents and items.
* Adds automatic holding creation for new items based on item type and item location.
* Adds automatic attaching of new items to a holding record if a valid holding record exists.
* Adds automatic holding deletion once last item is deleted.
* Adds automatic relinking if necessary to a new holding with item is updated.
* Adds loading fixtures is now using reroils class instead of invenio record class.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

# How to Test:
bootstrap
setup

1. create an item then check the json of the item to see if there is a holding field is added.
2. you can display the holding record via '/api/holdings/XX'
3. delete the  item and make sure the holding is also deleted by checking   '/api/holdings/XX'
